validator/evaluation/docker_evaluation.py

```diff
@@ -61,55 +88,29 @@ async def run_evaluation_docker(
             ],
             detach=True,
         )
-        # NOTE: replace with asyncio.to_thread
 
-        log_thread = threading.Thread(target=stream_logs, args=(container,))
-        log_thread.start()
-
-        result = container.wait()
-
-        log_thread.join()
+        log_task = asyncio.create_task(async_stream_logs(container))
```

Comment:
we're creating a task here for something which is already async. you can just await it. Better though is to get rid of the async_stream_logs function, and just define it as a syncronous function. 

Then create the task from the sync function